src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,518 @@
+    ///
+    pub fn new(model_type: ModelType, model: (&str,&str), config: (&str,&str), vocab: (&str,&str), merges: Option<(&str,&str)>) -> TokenClassificationConfig {
+        TokenClassificationConfig {
+            model_type: model_type,
+            model_resource: Resource::Remote(RemoteResource::from_pretrained(model)),
+            config_resource: Resource::Remote(RemoteResource::from_pretrained(config)),
+            vocab_resource: Resource::Remote(RemoteResource::from_pretrained(vocab)),
+            merges_resource: merges.map(|t| { Resource::Remote(RemoteResource::from_pretrained(t)) }),
+            device: Device::cuda_if_available(),
+        }
+    }
+}
+
+impl Default for TokenClassificationConfig {
+    /// Provides a default CONLL-2003 NER model (English)
+    fn default() -> TokenClassificationConfig {
+        TokenClassificationConfig {
+            model_type: ModelType::Bert,
+            model_resource: Resource::Remote(RemoteResource::from_pretrained(BertModelResources::BERT_NER)),
+            config_resource: Resource::Remote(RemoteResource::from_pretrained(BertConfigResources::BERT_NER)),
+            vocab_resource: Resource::Remote(RemoteResource::from_pretrained(BertVocabResources::BERT_NER)),
+            merges_resource: None,
+            device: Device::cuda_if_available(),
+        }
+    }
+}
+
+#[derive(Clone,Copy,Serialize,Deserialize)]
+/// # Identifies the type of model
+pub enum ModelType {
+    Bert,
+    DistilBert,
+    Roberta,
+}
+
+/// # Abstraction that holds a model configuration, can be of any of the supported models
+pub enum ConfigOption {
+    /// Bert configuration
+    Bert(BertConfig),
+    /// DistilBert configuration
+    DistilBert(DistilBertConfig),
+}
+
+/// # Abstraction that holds a particular tokenizer, can be of any of the supported models
+pub enum TokenizerOption {
+    /// Bert Tokenizer
+    Bert(BertTokenizer),
+    /// Roberta Tokenizer
+    Roberta(RobertaTokenizer),
+}
+
+/// # Abstraction that holds one particular token sequence classifier model, for any of the supported models
+pub enum TokenClassificationOption {
+    /// Bert for Token Classification
+    Bert(BertForTokenClassification),
+    /// DistilBert for Token Classification
+    DistilBert(DistilBertForTokenClassification),
+    /// Roberta for Token Classification
+    Roberta(RobertaForTokenClassification),
+}
+
+impl ConfigOption {
+    /// Interface method to load a configuration from file
+    pub fn from_file(model_type: ModelType, path: &Path) -> Self {
+        match model_type {
+            ModelType::Bert | ModelType::Roberta => ConfigOption::Bert(BertConfig::from_file(path)),
+            ModelType::DistilBert => ConfigOption::DistilBert(DistilBertConfig::from_file(path))
+        }
+    }
+
+    pub fn take_label_mapping(self) -> LabelMapping {
```

Comment:
I believe this could be simplified following harmonization of the `id2label` types (see comment below)